### PR TITLE
Allow configuring additional certificate authorities for a Hive instance

### PR DESCRIPTION
**Pull request by csrwng**

### CODE REVIEW

**Comment by csrwng** [comment]
WIP is pending manual testing

File: pkg/operator/hive/hive.go [hunk 1]
```diff
@@ -31,14 +33,17 @@ import (
...
 	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/client-go/kubernetes/scheme"
 )
 
 const (
-	dnsServersEnvVar = "ZONE_CHECK_DNS_SERVERS"
+	dnsServersEnvVar       = "ZONE_CHECK_DNS_SERVERS"
+	hiveAdditionalCASecret = "hive-additional-ca"
```
Comment (dgoodwin) [inline pkg/operator/hive/hive.go]: Probably worth godoc that this secret contains all the HiveConfig additional CAs assembled into one.

**Comment by csrwng** [comment]
@dgoodwin Manual testing is successful, removing the WIP
Also addressed godoc comments. 
For certs to work, we also need to cancel hold on https://github.com/openshift/hive/pull/388

**Pull request merged**